spec/workers/detect_invariants_daily_check_spec.rb

```diff
@@ -156,6 +157,23 @@
       )
     end
 
+    it 'detects application choices with out-of-date provider_ids' do
+      accredited_course = create(:course, :with_accredited_provider)
+      accredited_option = create(:course_option, course: accredited_course)
+      reverse_order = create(:application_choice, course_option: accredited_option)
+      reverse_order.update(provider_ids: reverse_order.provider_ids.reverse)
+
+      choices = create_list(:application_choice, 3).append(reverse_order)
+      choices.second.update(provider_ids: [])
+      choices.third.update(provider_ids: [1024])
+
+      described_class.new.perform
+
+      message = "#{choices[1].id}, #{choices[2].id}" # reverse order ignored
```

Comment:
could we assign the second/third applications to variables when updating them as worried of flakey tests